incubator/node/templates/deployment.yaml

```diff
@@ -17,9 +17,9 @@ spec:
     spec:
       initContainers:
       - name: git-clone-app
-        image: {{ .Values.devImage }}
+        image: {{ .Values.gitImage }}
         imagePullPolicy: {{ .Values.imagePullPolicy }}
-        command: [ '/bin/sh', '-c' , 'git clone {{ .Values.repository }} /app && git checkout {{ .Values.revision }}']
+        command: ['git', 'clone', '{{ .Values.repository }}', '-b', '{{ .Values.revision }}', '/app']
```

Comment:
`-b` argument for `git clone` will not work for revisions. It should either be a tag or branch.